pkg/metrics/otel.go

```diff
@@ -54,13 +54,18 @@ exporters:
   googlecloud/kubernetes:
     metric:
       prefix: "kubernetes.io/internal/addons/config_sync/"
-      skip_create_descriptor: false
+      skip_create_descriptor: true
+      instrumentation_library_labels: false
+      create_service_timeseries: true
+      service_resource_labels: false
     retry_on_failure:
       enabled: false
     sending_queue:
       enabled: false
 processors:
   batch:
+  resourcedetection:
+    detectors: [env, gcp]
```

Comment:
Do we still need the `resourcedetection` processor in https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/manifests/otel-agent-cm.yaml#L38-L39?

https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/manifests/templates/otel-collector.yaml#L26-L45 does not have the `resourcedetection` processor.

Is it necessary to use the `resourcedetection` processor in `otel-collector` instead of `otel-agent`?

Reply:
From the test I've been doing so far this `resourcedetection` in Otel Collector (the shared deployment) is necessary to correctly mirror the resource labels from OpenCensus to OpenTelemetry style. Haven't tried modifying the Otel Agent, but since the resource labels were originally added there, the `resourcedetection` should be kept.

https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/manifests/templates/otel-collector.yaml#L26-L45 This one is the default configuration that does not connect with Google Cloud APIs. When this configuration is in use, it usually mean user is not on a GKE cluster / has no access to Google Cloud API. In this case the `otel-collector` deployment only exports to the Prometheus endpoint, and with `resource_to_telemetry_conversion` it could carry CS resource labels as is.

Reply:
> From the test I've been doing so far this resourcedetection in Otel Collector (the shared deployment) is necessary to correctly mirror the resource labels from OpenCensus to OpenTelemetry style. Haven't tried modifying the Otel Agent, but since the resource labels were originally added there, the resourcedetection should be kept.

Can you add this a comment in the file?

Reply:
Done!